website/docs/r/redis_enterprise_database.html.markdown

```diff
@@ -50,8 +69,18 @@ The following arguments are supported:
 
 * `eviction_policy` - (Optional) Redis eviction policy - default is VolatileLRU. Possible values are `AllKeysLFU`, `AllKeysLRU`, `AllKeysRandom`, `VolatileLRU`, `VolatileLFU`, `VolatileTTL`, `VolatileRandom` and `NoEviction`. Defaults to `VolatileLRU`. Changing this forces a new Redis Enterprise Database to be created.
 
+-> **NOTE:** Eviction Policy must be set to NoEviction when using RediSearch module
```

Comment:
is there a code check for this?

Reply:
yes, there is a logic to check the module restriction and the eviction policy:

```
module:
if moduleName != "RediSearch" && isGeoEnabled {
			return nil, fmt.Errorf("Only RediSearch module is allowed with geo-replication")
		}

eviction policy:
if isGeoEnabled && module != nil && evictionPolicy != databases.EvictionPolicyNoEviction {
		return fmt.Errorf("evictionPolicy must be set to NoEviction when using RediSearch module")
	}

```